scilpy/segment/recobundlesx.py

```diff
@@ -232,40 +229,40 @@ def _register_model_to_neighb(self, slr_num_thread=1,
                 slr = StreamlineLinearRegistration(metric=metric,
                                                    x0=init_transfo_dof,
                                                    bounds=bounds_dof[:9],
-                                                   num_threads=slr_num_thread)
+                                                   num_threads=1)
                 slm = slr.optimize(static, moving)
         self.model_centroids = transform_streamlines(self.model_centroids,
                                                      np.linalg.inv(slm.matrix))
 
-    def prune_far_from_model(self, bundle_pruning_thr=10,
-                             neighbors_cluster_thr=8):
+    def prune_far_from_model(self, pruning_thr=10):
         """
         Wrapper function to prune clusters from the tractogram too far from
         the model.
-        :param neighbors_to_prune, list or arraySequence, streamlines to prune.
-        :param bundle_pruning_thr, float, distance in mm for pruning.
-        :param neighbors_cluster_thr, float, distance in mm for clustering.
+        :param pruning_thr, float, distance in
+            thresholds = [32, 16, 24, neighbors_cluster_thr]
         """
         # Neighbors can be refined since the search space is smaller
-        thresholds = [32, 16, 24, neighbors_cluster_thr]
+        t0 = time()
 
-        neighb_cluster_map = qbx_and_merge(self.neighb_streamlines, thresholds,
-                                           nb_pts=self.nb_points,
-                                           rng=self.rng, verbose=False)
+        neighb_streamlines = reconstruct_streamlines_from_memmap(
+            self.memmap_filenames, self.neighb_indices, strs_dtype=np.float16)
 
-        dist_matrix = bundles_distances_mdf(self.model_centroids,
-                                            neighb_cluster_map.centroids)
-        dist_matrix[np.isnan(dist_matrix)] = np.inf
-        dist_matrix[dist_matrix > bundle_pruning_thr] = np.inf
-        mins = np.min(dist_matrix, axis=0)
+        with warnings.catch_warnings(record=True) as _:
+            fss = FastStreamlineSearch(neighb_streamlines,
+                                       pruning_thr+1, resampling=12)
```

Comment:
Don't forget to change the "pruning_thr+1" to "pruning_thr" !
This should reduce the computation time, without precision loss.